datastore/index.js

```diff
@@ -1,29 +1,45 @@
+const fs = require('fs');
+
+let _data = { 'subscriptions': [] };
+const _datastore_path = 'db.json';
+
 module.exports = {
-  async add({ channelId, githubProjectUrl }) {
-    // TODO save to database
-    console.log(`Subscibe ${channelId} to ${githubProjectUrl}`);
+  async connectToDatastore() {
+    fs.readFile(_datastore_path, (err, data) => {
+      if (err) {
+        if (err.code === 'ENOENT') {
+          // Create local datastore
+          module.exports.commit();
+        }
+        else {
+          throw err;
+        }
+      }
+      else {
+        _data = JSON.parse(data);
+      }
+    });
   },
-  async remove({ channelId, githubProjectUrl }) {
-    // TODO remove from database
-    console.log(`Unsubscribe ${channelId} to ${githubProjectUrl}`);
+  add({ channelId, githubProjectUrl, commit = true }) {
+    _data.subscriptions.push({ channelId: channelId, githubProjectUrl: githubProjectUrl });
```

Comment:
Need to check for duplications before pushing.

Reply:
Instead of what we originally had `queryByGitHubProjectUrl ` and `queryByChannelId `

Maybe we can create a single query method that takes in both:
```
query({ channelId, githubProjectUrl }) {
  return _data.subscriptions.filter(subscription => 
      (channelId ? subscription.channelId === channelId : true) &&
      (githubProjectUrl ? subscription.githubProjectUrl === githubProjectUrl : true)
  )
}
```

Just an idea~